lib/serverUtils.js

```diff
@@ -206,24 +206,118 @@ const deleteAccount = function (req, bcrypt, postgres, callback) {
                 });
             });
         }
-        else{
+        else {
             callback(null, -1);
         }
     });
 }
 
+const getCartFromDatabase = function (userId, postgres, callback) {
+    // -1 - empty cart else [0:-2] - items [-1] - price
+    postgres.query(`SELECT * FROM orders WHERE userid=${userId} and orderstatus=0`, function (err, resultOrder) {
+        if (err) {
+            throw (err);
+        }
+        if (!resultOrder.rows[0]) {
+            callback(null, -1);
+        }
+        else {
+            let price = parseInt(resultOrder.rows[0].valueoforder)
+            let orderId = parseInt(resultOrder.rows[0].id)
+            postgres.query(`SELECT * FROM ordereditems WHERE orderid=${orderId}`), function (err, result) {
+                if (err) {
+                    throw (err);
+                }
+                callback(null, { orderedItems: result, valueOfOrder: price});
+            }
+        }
+    })
+}
+
+const addItemToCartTable = function (productId, orderId, postgres, callback) {
+    // check if product in cart
+    postgres.query(`SELECT * from ordereditems where orderid=${orderId} and itemid=${productId}`, function (err, resultOrder) {
+        if (err) {
+            throw (err);
+        }
+        //no product in cart
+        if (!resultOrder.rows[0]) {
+            postgres.query(`INSERT INTO ordereditems (orderid, itemid, amount) 
+                            VALUES (${orderId}, ${productId}, 1)`, function (err, result) {
+                if (err) {
+                    throw (err);
+                }
+            })
+        }
+        // product in cart
+        else {
+            postgres.query(`UPDATE ordereditems SET amount = amount + 1 where orderid = ${orderId} and itemid=${productId}`, function (err, result) {
+                if (err) {
+                    throw (err);
+                }
+            })
+        }
+        callback(null, 1);
```

Comment:
Pytanie czy ten callback nie powinien się wykonać dopiero po update? Bo w tej strukturze nie mamy pewności, że update wykona się przed wykonaniem funkcji callback. Dlatego warto było by to przerzucić do funkcji która uruchamia się po update.